src/main/scala/com/databricks/spark/redshift/RedshiftWriter.scala

```diff
@@ -311,29 +311,25 @@ private[redshift] class RedshiftWriter(
       // for a description of the manifest file format. The URLs in this manifest must be absolute
       // and complete.
 
-      // The saved filenames are going to be of the form part-r-XXXXX-UUID.avro. There isn't a way
-      // to figure out what this UUID is, besides querying S3. In principle, this file format could
-      // change in a way that breaks this library, but this seems unlikely and we can patch around
-      // it if that happens.
+      // The saved filenames depend on the spark-avro version. In spark-avro 1.0.0, the write
+      // path uses SparkContext.saveAsHadoopFile(), which produces filenames of the form
+      // part-XXXXX.avro. In spark-avro 2.0.0+, the partition filenames are of the form
+      // part-r-XXXXX-UUID.avro.
       val fs = FileSystem.get(URI.create(tempDir), sqlContext.sparkContext.hadoopConfiguration)
-      val writeJobUUID = fs.listStatus(new Path(tempDir))
-        .iterator
-        .map(_.getPath.getName)
-        .filter(_.startsWith("part"))
-        .take(1)
-        .toSeq
-        .headOption.getOrElse(throw new Exception("No part files were written!"))
-        .drop("part-r-XXXXX-".length)
-        .stripSuffix(".avro")
-
+      val partitionIdRegex = "^part-(?:r-)?(\\d{5}).*$".r
```

Comment:
It just occurred to me: what if you have more than 99999 partitions? What format does the filename have in that case?